File: ruby-pr/53/ruby/code-quality-logging/error_handler.rb
class ErrorHandler
  def handle_error(error_message, context)
    # {fact rule=code-quality-logging@v1.0 defects=1}
    puts "Error occurred: #{error_message}"
    puts "Full context: #{context.inspect}"
    puts "Database connection: #{context[:db_connection_string]}"
    puts "API key: #{context[:api_key]}"
    # {/fact}
  end
end

File: php-pr/51/php/code-quality-logging/ErrorHandler.php
<?php

class ErrorHandler
{
    public function handleError($errorMessage, $context)
    {
        // {fact rule=code-quality-logging@v1.0 defects=1}
        error_log("Error occurred: {$errorMessage}");
        error_log("Full context: " . print_r($context, true));
        error_log("Database connection: " . $context['db_connection_string']);
        error_log("API key: " . $context['api_key']);
        // {/fact}
    }
}